tensorflow/core/grappler/optimizers/auto_mixed_precision_test.cc

```diff
@@ -1095,8 +1074,8 @@ TEST_F(AutoMixedPrecisionTest, BatchMatMul) {
 #if GOOGLE_CUDA
   if (GetCudaVersion(*virtual_cluster_.get()) >= 9010) {
 #elif TENSORFLOW_USE_ROCM
-  if (HasEnhancedFP16ComputeSupport(GetMinDeviceGPUArch(*virtual_cluster_.get()))){
-#else  // TENSORFLOW_USE_ROCM
+  if (HasFP16Support(*virtual_cluster_.get())){
+#else  
```

Comment:
Once you make the change suggested above to have this test always run with `gfx906`, this check becomes moot, and you can revert to having `if (true)` as before.  
* It also means the `HasFP16Support` routine is not needed, which then implies
  * there will no more references to `GetFastFP16Support` in this file, which then implies
    * `GetFastFP16Support` can be kept local to the `auto_mixed_precision.cc` file!